lighthouse-core/gather/gatherers/obscuring-elements.js

```diff
@@ -0,0 +1,109 @@
+const pageFunctions = require('../../lib/page-functions.js');
+const TraceProcessor = require('../../lib/tracehouse/trace-processor.js');
+const { getOuterHTMLSnippet } = require('../../lib/page-functions.js');
+
+/** @typedef {{element?: string, x?: number, y?:number, retList?: Array<any>}} MyData */
+/** @typedef {{xCoord: number, yCoord: number}} Point */
+
+/**
+ * @this {HTMLElement}
+ * @return {LH.Artifacts['ObscuringElements']}
+ */
+/* istanbul ignore next */
+function checkIntersection() {
+  const lcpElement = this.nodeType === document.ELEMENT_NODE ? this : this.parentElement; // eslint-disable-line no-undef
+  const lcpHTMLSnippet = lcpElement && getOuterHTMLSnippet(lcpElement);
+  const obscuringElements = [];
+  const seen = new Set();
+  /**
+   * @param {HTMLElement | Element} el 
+   */
+  function getChildrenSnippets(el) {
+    if (!el) return;
+    const snippet = getOuterHTMLSnippet(el);
+    seen.add(snippet);
+    for (const child of el.children) {
+      getChildrenSnippets(child);
+    }
+  }
+  lcpElement && getChildrenSnippets(lcpElement);
+  const boundingRect = lcpElement && lcpElement.getBoundingClientRect();
+  if (!boundingRect) return {};
+  const {x, y, width, height} = boundingRect;
+  /** @type {Array<Point>} */
+  const points = [];
+  for (let i = 0.1; i < 1.0; i += 0.2) {
+    for (let j = 0.1; j < 1.0; j += 0.2) {
+      const xCoord = x + Math.round(width * i);
+      const yCoord = y + Math.round(height * j);
+      points.push({xCoord, yCoord});
+    }
+  }
+
+  for (const {xCoord, yCoord} of points) {
+    for (const element of document.elementsFromPoint(xCoord, yCoord)) {
+      // @ts-ignore - put into scope via stringification
+      const snippet = getOuterHTMLSnippet(element); // eslint-disable-line no-undef
+      if (snippet === lcpHTMLSnippet) break;
+      if (seen.has(snippet))
+        continue;
+      seen.add(snippet);
+      const rect = element.getBoundingClientRect();
+      obscuringElements.push({x: rect.x, y: rect.y, width: rect.width, height: rect.height, snippet});
+    }
+  }
+
+  return obscuringElements;
+}
+
+class ObscuringElements extends Gatherer {  
+  /**
+   * @param {LH.Gatherer.PassContext} passContext
+   * @param {LH.Gatherer.LoadData} loadData
+   * @return {Promise<LH.Artifacts['ObscuringElements'] | null>}
+   */
+  async afterPass(passContext, loadData) {
+    const driver = passContext.driver;
+    if (!loadData.trace) {
+      throw new Error('Trace is missing!');
+    }
+
+    const {largestContentfulPaintEvt} = TraceProcessor.computeTraceOfTab(loadData.trace);
+    const backendNodeId = largestContentfulPaintEvt && largestContentfulPaintEvt.args &&
+      largestContentfulPaintEvt.args.data && largestContentfulPaintEvt.args.data.nodeId;
+
+    if (!backendNodeId) return null;
+    const objectId = await driver.resolveNodeIdToObjectId(backendNodeId);
+    const response = await driver.sendCommand('Runtime.callFunctionOn', {
```

Comment:
evaluateAsync has a bunch of error handling awesomeness, and i think we want to use it here, too. (it'd also take care of the extra value resolving on L100)

we have two places we also want to a variable into a page fn: 

https://github.com/GoogleChrome/lighthouse/blob/259e2228ac8131ccc43ae0fbb3670adb2cbc451e/lighthouse-core/gather/gatherers/seo/tap-targets.js#L309
https://github.com/GoogleChrome/lighthouse/blob/259e2228ac8131ccc43ae0fbb3670adb2cbc451e/lighthouse-core/gather/fetcher.js#L178

so we can do that to and your L93 would become something like 

```js
return checkIntersection(${JSON.stringify(objectId)});
```

Reply:
The only reason I stuck with `Runtime.callFunctionOn` was because I couldn't figure out a simple way to get to the element with just the nodeId. The only alternative I know of is setting a temp attribute and querying for it like we did initially with TraceElements.

Reply:
fyi #10816 makes an interface in Driver for `callFunctionOn` just like `evalAsync`.

Reply:
How close is option 1 to being done?